include/range/v3/view_adaptor.hpp

```diff
@@ -80,7 +80,11 @@ namespace ranges
               : compressed_pair<BaseIter, Adapt>
             {
                 using compressed_pair<BaseIter, Adapt>::compressed_pair;
+#ifdef RANGES_WORKAROUND_MSVC_688606
+                using value_type = value_type_t<Adapt>;
```

Comment:
This one is my favorite: since `Adapt` is an indirect base, MSVC effectively rewrites line 86 to `using value_type = typename compressed_pair<BaseIter, Adapt>::ebo_box<Adapt>::Adapt::value_type;`, and then complains that this class can't access that member because `ebo_box` is a private base.

Reply:
🤢